processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -77,6 +79,8 @@
     DECORATOR_CONSTRUCTOR( "Specified decorator type has no default constructor nor a constructor with a single parameter accepting the decorated mapper type." ),
 
     GENERAL_NO_IMPLEMENTATION( "No implementation type is registered for return type %s." ),
+    GENERAL_ABSTRACT_RETURN_TYPE( "The return type %s is abstract. Provide non abstract result type or a factory method." ),
+    GENERAL_INTERFACE_RETURN_TYPE( "The return type %s is interface. Provide non abstract/interface result type or a factory method." ),
```

Comment:
Seems "non abstract" is a left-over? I'd handle both cases together as lined out above.